### PR TITLE
ISSUE-495 Implement FirebaseSubscription/get

### PR DESCRIPTION
**Pull request by vttranlina**
resolve: https://github.com/linagora/tmail-backend/issues/495

### CODE REVIEW

**Comment by quantranhong1999** [comment]
```
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-surefire-plugin:2.22.2:test (default-cli) on project memory-jmap-integration-tests: There are test failures.

[Guice/MissingImplementation]: No implementation for FirebaseSubscriptionRepository was bound.
```

Binded here: https://github.com/linagora/tmail-backend/pull/517/commits/b85135f73726bc2781fbc60440acf6f860a689a7

File: tmail-backend/integration-tests/jmap/jmap-integration-tests-common/src/main/scala/com/linagora/tmail/james/common/FirebaseSubscriptionGetMethodContract.scala
Comment (Arsnael) [inline tmail-backend/integration-tests/jmap/jmap-integration-tests-common/src/main/scala/com/linagora/tmail/james/common/FirebaseSubscriptionGetMethodContract.scala:408]:
```suggestion
  def getShouldReturnNotFoundWhenDoesNotHavePermission(server: GuiceJamesServer): Unit = {
```

File: tmail-backend/integration-tests/jmap/jmap-integration-tests-common/src/main/scala/com/linagora/tmail/james/common/FirebaseSubscriptionGetMethodContract.scala
Comment (Arsnael) [inline tmail-backend/integration-tests/jmap/jmap-integration-tests-common/src/main/scala/com/linagora/tmail/james/common/FirebaseSubscriptionGetMethodContract.scala:452]: quotaGet ? :)

```suggestion
  def getShouldReturnOnlyRequestedProperties(server: GuiceJamesServer): Unit = {
```

**Review by Arsnael — COMMENTED** [review]
Let's not forget to add the distributed test when cassandra api is merged as well

File: tmail-backend/jmap/extensions/src/main/scala/com/linagora/tmail/james/jmap/json/FirebaseSubscriptionSerializer.scala
Comment (chibenwa) [inline tmail-backend/jmap/extensions/src/main/scala/com/linagora/tmail/james/jmap/json/FirebaseSubscriptionSerializer.scala:23]: Shouldn't we use UTCDate in the data model instead?

File: tmail-backend/jmap/extensions/src/main/scala/com/linagora/tmail/james/jmap/method/FirebaseSubscriptionGetMethod.scala
Comment (chibenwa) [inline tmail-backend/jmap/extensions/src/main/scala/com/linagora/tmail/james/jmap/method/FirebaseSubscriptionGetMethod.scala:35]: No @Inject ?

Comment (vttranlina) [reply tmail-backend/jmap/extensions/src/main/scala/com/linagora/tmail/james/jmap/method/FirebaseSubscriptionGetMethod.scala:35]: I created it manually without annotation, (because we have a enable/disable firebase feature)

File: tmail-backend/jmap/extensions/src/main/scala/com/linagora/tmail/james/jmap/method/FirebaseSubscriptionGetMethod.scala
Comment (quantranhong1999) [inline tmail-backend/jmap/extensions/src/main/scala/com/linagora/tmail/james/jmap/method/FirebaseSubscriptionGetMethod.scala:38]: IMO just use normal @Inject and guice binding. The module chooser will take care of the `enable/disable firebase feature`.

Comment (vttranlina) [reply tmail-backend/jmap/extensions/src/main/scala/com/linagora/tmail/james/jmap/method/FirebaseSubscriptionGetMethod.scala:38]: repository, method... always create instance when runtime even when firebase feature is disable?

Comment (quantranhong1999) [reply tmail-backend/jmap/extensions/src/main/scala/com/linagora/tmail/james/jmap/method/FirebaseSubscriptionGetMethod.scala:38]: > The module chooser will take care of the enable/disable firebase feature

That means the module chooser already takes care of whether loading the beans at runtime.

File: tmail-backend/jmap/extensions/src/main/scala/com/linagora/tmail/james/jmap/method/FirebaseSubscriptionGetMethod.scala
Comment (chibenwa) [reply tmail-backend/jmap/extensions/src/main/scala/com/linagora/tmail/james/jmap/method/FirebaseSubscriptionGetMethod.scala:42]: Where?

Just add the module inside the module chooser and rely on standard injects please.

File: tmail-backend/jmap/extensions/src/main/java/com/linagora/tmail/james/jmap/firebase/FirebaseModule.java
Comment (quantranhong1999) [inline tmail-backend/jmap/extensions/src/main/java/com/linagora/tmail/james/jmap/firebase/FirebaseModule.java:12]: Please remove this memory module. This is not a common module (for both memory and distributed TMail), this variant would be taken care of by the module chooser instead.

see https://github.com/linagora/tmail-backend/pull/517/commits/b3da7455926254895c652c5f7c433dd228c36600

Comment (vttranlina) [reply tmail-backend/jmap/extensions/src/main/java/com/linagora/tmail/james/jmap/firebase/FirebaseModule.java:12]: if it is removed now, ci will fail. 
or waiting until #517 has been merged, then I will rebase later

Comment (quantranhong1999) [reply tmail-backend/jmap/extensions/src/main/java/com/linagora/tmail/james/jmap/firebase/FirebaseModule.java:12]: Merged. Please rebase.

**Review by chibenwa — APPROVED** [review]

**Review by Arsnael — APPROVED** [review]

**Pull request merged**
